tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala

```diff
@@ -0,0 +1,125 @@
+import io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE
+import io.netty.handler.codec.http.HttpResponseStatus.{BAD_REQUEST, FORBIDDEN, INTERNAL_SERVER_ERROR, UNAUTHORIZED}
+import io.netty.handler.codec.http.{HttpMethod, HttpResponseStatus, QueryStringDecoder}
+import org.apache.james.jmap.HttpConstants.JSON_CONTENT_TYPE
+import org.apache.james.jmap.core.ProblemDetails
+import org.apache.james.jmap.exceptions.UnauthorizedException
+import org.apache.james.jmap.http.Authenticator
+import org.apache.james.jmap.json.ResponseSerializer
+import org.apache.james.jmap.routes.ForbiddenException
+import org.apache.james.jmap.{Endpoint, JMAPRoute, JMAPRoutes}
+import org.apache.james.metrics.api.MetricFactory
+import play.api.libs.json.Json
+import reactor.core.scala.publisher.SMono
+import reactor.core.scheduler.Schedulers
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
+
+import java.nio.charset.StandardCharsets
+import java.time.{Clock, Duration, ZonedDateTime}
+import java.util.stream
+import javax.inject.Inject
+import scala.jdk.CollectionConverters._
+
+case class JwtTokenRoutesModule() extends AbstractModule {
+  override def configure(): Unit = {
+    Multibinder.newSetBinder(binder, classOf[JMAPRoutes])
+      .addBinding().to(classOf[JwtTokenRoutes])
+  }
+
+  @Provides
+  @Singleton
+  @Named(LongLivedTokenInjectKeys.JMAP)
+  def provideLongLivedTokenAuthenticator(metricFactory: MetricFactory,
+                                         longLivedTokenAuthenticationStrategy: LongLivedTokenAuthenticationStrategy): Authenticator =
+    Authenticator.of(metricFactory, longLivedTokenAuthenticationStrategy)
+
+}
+
+case class JwtTokenResponse(token: JWTToken, expiresOn: ZonedDateTime)
+
+object JwtTokenRoutes {
+  val ENDPOINT: String = "token"
+  val TOKEN_DURATION_DEFAULT: Duration = Duration.ofHours(1)
+  val TYPE_PARAM: String = "type"
+  val DEVICE_ID_PARAM: String = "deviceId"
+  val TYPE_ACCEPT_PARAM: String = "shortLived"
+}
+
+class JwtTokenRoutes @Inject()(@Named(LongLivedTokenInjectKeys.JMAP) val authenticator: Authenticator,
+                               jwtSigner: JwtSigner,
+                               clock: Clock) extends JMAPRoutes {
+  override def routes(): stream.Stream[JMAPRoute] = stream.Stream.of(
+    JMAPRoute.builder
+      .endpoint(new Endpoint(HttpMethod.GET, s"/$ENDPOINT"))
+      .action(this.generate)
+      .corsHeaders())
+
+  private def generate(request: HttpServerRequest, response: HttpServerResponse): SMono[Unit] =
+    SMono(authenticator.authenticate(request))
+      .doOnNext(_ => validRequestParameter(request))
+      .map(session => {
+        val validUntil: ZonedDateTime = ZonedDateTime.ofInstant(clock.instant().plusSeconds(TOKEN_DURATION_DEFAULT.toSeconds), clock.getZone)
+        jwtSigner.sign(session.getUser, validUntil) -> validUntil
+      })
+      .map(tokenAndValidUntil => JwtTokenResponse(tokenAndValidUntil._1, tokenAndValidUntil._2))
```

Comment:
```suggestion
      .map(session => {
        val validUntil: ZonedDateTime = ZonedDateTime.ofInstant(clock.instant().plusSeconds(TOKEN_DURATION_DEFAULT.toSeconds), clock.getZone)
         JwtTokenResponse(jwtSigner.sign(session.getUser, validUntil), validUntil)
      })
```

Get rid of the unneeded tuple?